tests/test_aaa_authentication_login.rb

```diff
@@ -27,315 +28,199 @@ def aaaauthenticationlogin_detach(authlogin)
+  def test_get_default_ascii_authentication
     aaaauthlogin = AaaAuthenticationLogin
-    s = @device.cmd("configure terminal")
-    s = @device.cmd("no aaa authentication login ascii-authentication")
-    s = @device.cmd("end")
-    # Flush the cache since we've modified the device
-    node.cache_flush
+    config('no aaa authentication login ascii-authentication')
     assert_equal(DEFAULT_AAA_AUTHENTICATION_LOGIN_ASCII_AUTH,
                  aaaauthlogin.default_ascii_authentication,
-                 "Error: AAA authentication login, default ascii incorrect")
+                 'Error: AAA authentication login, default ascii incorrect')
     aaaauthenticationlogin_detach(aaaauthlogin)
   end
 
-  def test_aaaauthenticationlogin_set_ascii_authentication
-    state = true
-
+  def test_set_ascii_authentication
     aaaauthlogin = AaaAuthenticationLogin
 
-    aaaauthlogin.ascii_authentication = state
-    line = get_match_line("ascii-authentication")
-    refute_nil(line, "Error: AAA authentication login ascii not configured #{state}")
-    assert(aaaauthlogin.ascii_authentication,
-                 "Error: AAA authentication login asci not set #{state}\n" +
-                 "See CSCuu12667 (4/29/15)")
-
-    # Now bring it back to default
-    state = DEFAULT_AAA_AUTHENTICATION_LOGIN_ASCII_AUTH
-    aaaauthlogin.ascii_authentication = state
-    line = get_match_line("ascii-authentication")
-    refute_nil(line, "Error:  AAA authentication login, default ascii not configured")
-    refute(aaaauthlogin.ascii_authentication,
-                 "Error: AAA authentication login asci not set #{state}\n" +
-                 "See CSCuu12667 (4/29/15)")
+    aaaauthlogin.ascii_authentication = true
+    assert_show_match(command: 'show run aaa all | no-more',
+                      pattern: /^aaa authentication login ascii-authentication/)
+
+    aaaauthlogin.ascii_authentication =
+      DEFAULT_AAA_AUTHENTICATION_LOGIN_ASCII_AUTH
+    refute_show_match(command: 'show run aaa all | no-more',
+                      pattern: /^aaa authentication login ascii-authentication/)
 
     aaaauthenticationlogin_detach(aaaauthlogin)
   end
 
-  def test_aaaauthenticationlogin_get_chap
+  def test_get_chap
     aaaauthlogin = AaaAuthenticationLogin
 
-    s = @device.cmd("configure terminal")
-    s = @device.cmd("no aaa authentication login chap enable")
-    s = @device.cmd("end")
-    # Flush the cache since we've modified the device
-    node.cache_flush
+    config('no aaa authentication login chap enable')
     refute(aaaauthlogin.chap,
-                 "Error: AAA authentication login chap get\n" +
-                 "See CSCuu12667 (4/29/15)")
-
-    s = @device.cmd("configure terminal")
-    s = @device.cmd("aaa authentication login chap enable")
-    s = @device.cmd("end")
-    # Flush the cache since we've modified the device
-    node.cache_flush
+           "Error: AAA authentication login chap get\n" \
+           'See CSCuu12667 (4/29/15)')
```

Comment:
CSCuu12667 is fixed and was not in any released code - this reference should be removed.